src/apdu_sign.c

```diff
@@ -285,7 +285,9 @@ size_t handle_apdu_sign_with_hash(uint8_t instruction, volatile uint32_t *flags)
 
 int perform_signature(bool const on_hash, bool const send_hash) {
     write_high_water_mark(&G.parsed_baking_data);
-
+    if (global.is_blank_screen == false) {
+        ux_empty_screen();
+    }
```

Comment:
Maybe it could be better to check this condition in the `ux_empty_screen` itself

Reply:
Done.